src/lib/transaction_snark/transaction_snark.mli

```diff
@@ -3,7 +3,15 @@ open Coda_base
 open Snark_params
 
 module Proof_type : sig
-  type t = [`Merge | `Base] [@@deriving bin_io, sexp, yojson]
+  module Stable : sig
+    module V1 : sig
+      type t = [`Base | `Merge] [@@deriving bin_io, sexp, yojson]
```

Comment:
the order of the variants here is not the same. does the bin_io deriving care about the order in order to generate ids?

Reply:
I don't think it matters here. The section on polymorphic variants in https://github.com/janestreet/bin_prot, IIUC, that the tags are written as hashes of the tags themselves.

But I can change it, if you like, to eliminate that concern.

Reply:
I think it's fine.

Reply:
Normalized all occurrences so that `Base` is first.